unity-renderer/Assets/Scripts/MainScripts/DCL/Components/UI/UIShape.cs

```diff
@@ -153,7 +151,13 @@ public class Model : BaseModel
 
         public UIShape parentUIComponent { get; protected set; }
 
-        public UIShape() { model = new Model(); }
+        public UIShape()
+        {
+            DataStore.i.screen.size.OnChange += OnScreenResize;
```

Comment:
When working with the `DataStore` I find better to have a property in the class
`
private Vector2Int screenSize => DataStore.i.screen.size;
`

It reduces the amount of "noise" when searching for references when lot of systems are accessing a data.